jetstream/abouthome-startup-cache-3.toml

```diff
@@ -50,7 +50,34 @@ FROM (
 """
 
 ## Data Sources
+[data_sources]
 
+[data_sources.main]
+from_expression = """(
+    SELECT
+        *, 
+        DATE(submission_timestamp) AS submission_date,
+        environment.experiments
+    FROM `moz-fx-data-shared-prod`.telemetry_stable.main_v4
+    WHERE `mozfun.map.get_key`(
+        environment.experiments, 'abouthome-startup-cache-3'
+    ).branch IS NOT NULL
+
+)"""
+experiments_column_type = "native"
+friendly_name = "Main"
+description = "Main ping table"
+
+[data_sources.clients_daily]
+from_expression = """(
+        SELECT * FROM
+                mozdata.telemetry.clients_daily d
+        WHERE `mozfun.map.get_key`(
+                d.experiments, 'abouthome-startup-cache-3'
+        ) IS NOT NULL
+)"""
+friendly_name = "Clients Daily"
+description = "Clients Daily"
```

Comment:
@danielkberry  This should be ok, right? The query did finish when filtering out clients based on the experiment slug before doing the join

Reply:
This is a genius idea